tests/testthat/test-get_upw.R

```diff
@@ -61,20 +61,39 @@ test_that("unpaywall mining works",{
   expect_true("oa_color" %in% names(upwdf))
   # the data frame is as long as all DOIs
   expect_true(nrow(upwdf) == length(dois))
-  #add line for testing green over bronze prioritization
+  #add line for testing green over bronze prioritization once implemented
 })
 
-test_that("errors can be dealt with",{
-  doi_fails <- c(
+test_that("errors are handled correctly",{
+  doi_some_fails <- c(
+    "10.1007/jhep11(2015)127",
+    "10.1109/icdsp.2015.7251906",
+    "10.1117/1.jei.24.5.053018",
+    "10.1007/ 78-3-319-27857-5_z", #invalid DOI
+    "10.14273/unisa-1437" #DataCite DOI
+  )
+  df <- tibble::tibble(doi=doi_some_fails)
+  expect_output({
+    upwdf <- get_upw(df, email = "b.m.r.kramer@uu.nl")})
+  # the result is a data frame
+  expect_true("data.frame" %in% class(upwdf))
```

Comment:
These things are already tested in the test above. As mentioned, it is not a huge problem that there is overlap in tests, but I would keep this to the point, and remove the tests that check for data frame and existence of oa_color

Reply:
Good point - now removed